### PR TITLE
[Merged by Bors] - [Merged by Bors] - feat: pretriangulated categories have finite biproducts

### PR DESCRIPTION
**Pull request by joelriou**
---
- [x] depends on: #7643 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Triangulated/Pretriangulated.lean [hunk 1]
```diff
@@ -457,10 +458,78 @@ lemma isIso₁_of_isIso₂₃ {T T' : Triangle C} (φ : T ⟶ T') (hT : T ∈ di
...
+      simp only [sub_comp, assoc, id_comp, ← inr_snd, comp_id, sub_self])
+  refine' ⟨⟨binaryBiproductData _ mem rfl inr inr_snd.symm fst _⟩⟩
+  dsimp
+  simp only [← hfst, sub_add_cancel]⟩
+
+instance : HasFiniteProducts C := hasFiniteProducts_of_has_binary_and_terminal
+instance : HasFiniteCoproducts C := hasFiniteCoproducts_of_has_binary_and_initial
+instance : HasFiniteBiproducts C := HasFiniteBiproducts.of_hasFiniteProducts
+
+lemma exists_iso_binaryBiproduct_of_dist_triangle (T : Triangle C) (hT : T ∈ distTriang C)
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Triangulated/Pretriangulated.lean:506]: I think I would use more camelCase, or is there already some convention that I missed?
```suggestion
lemma exists_iso_binaryBiproduct_of_distTriangle (T : Triangle C) (hT : T ∈ distTriang C)
```

Comment (joelriou) [reply Mathlib/CategoryTheory/Triangulated/Pretriangulated.lean:506]: I also had some hesitation here. We also have many lemmas like `rot_of_dist_triangle`. Should we also rename these as `rot_of_distTriang`, etc?

Comment (joelriou) [reply Mathlib/CategoryTheory/Triangulated/Pretriangulated.lean]: I have changed `rot_of_dist_triangle` in #7643, and this lemma is now `exists_iso_binaryBiproduct_of_distTriang`.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**